### PR TITLE
Only add account index if not already present

### PR DESCRIPTION
**Pull request by Thoralf-M**
# Description of change

Only add account index if not already present, because it would otherwise lead to duplicated accounts

## Type of change

- Bug fix (a non-breaking change which fixes an issue)

## How the change has been tested

Examples

## Change checklist

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that new and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by Tuditi — DISMISSED** [review]
LGTM!

File: src/storage/manager.rs [hunk 1]
```diff
@@ -131,7 +131,11 @@ impl StorageManager {
     }
 
     pub async fn save_account(&mut self, account: &Account) -> crate::Result<()> {
-        self.account_indexes.push(*account.index());
+        // Only add account index if not already present
+        if !self.account_indexes.iter().any(|index| index == account.index()) {
```
Comment (thibault-martinez) [inline src/storage/manager.rs]: use `contains` instead ?

**Review by pelumi-ajayi — DISMISSED** [review]

File: src/storage/manager.rs (hunk 1 above)
Comment (Thoralf-M) [reply src/storage/manager.rs]: Wen Clippy saying me that?

Comment (thibault-martinez) [reply src/storage/manager.rs]: Yeah it definitely should :(

**Review by thibault-martinez — APPROVED** [review]

**Review by samuel-rufi — APPROVED** [review]

**Pull request merged**